tests/utils.py

```diff
@@ -265,19 +266,23 @@ def setup_test_homeserver(
         cur.close()
         db_conn.close()
 
+    def inject_cache(obj, kw: dict):
+        # Install @cache_in_self attributes
+        for key, val in kw.items():
+            setattr(obj, "_" + key, val)
+
+        obj.tls_server_context_factory = Mock()
+        obj.tls_client_options_factory = Mock()
+
     if datastore is None:
-        hs = homeserverToUse(
-            name,
-            config=config,
-            version_string="Synapse/tests",
-            tls_server_context_factory=Mock(),
-            tls_client_options_factory=Mock(),
-            reactor=reactor,
-            **kargs
+        hs = homeserver_to_use(
+            name, config=config, version_string="Synapse/tests", reactor=reactor,
         )
 
+        inject_cache(hs, kwargs)
+
         hs.setup()
-        if homeserverToUse.__name__ == "TestHomeServer":
+        if homeserver_to_use == TestHomeServer:
```

Comment:
Unless there is another class that is also exactly called `TestHomeServer` within the test framework, this is more idiomatic